components/event-publisher-proxy/pkg/handler/http/handler_test.go

```diff
@@ -460,14 +464,27 @@ func setupTestResources(t *testing.T, port, maxRequestSize int, applicationName,
 
 func validateApplicationName(appName string) testingutils.Validator {
 	return func(r *http.Request) error {
+		if r.Header.Get(cev2http.ContentType) == cloudevents.CEStructuredMode {
+			// CE structured mode
+			message := cev2http.NewMessageFromHttpRequest(r)
+			defer func() { _ = message.Finish(nil) }()
+			if event, err := binding.ToEvent(context.Background(), message); err != nil {
+				return err
+			} else if strings.Contains(event.Type(), appName) {
```

Comment:
Why is this check needed?

Can we somehow remove if...elseif...else block?

Reply:
The "appName" location is found in different places dependent of CE event (structured or binary). The test check both structured and binary events

Reply:
Refactored